### ISSUE TITLE
Events are not constrained to 8 outcomes yet many places in the code only work with the first 8.

### ISSUE THREAD

**Issue by MicahZoltu** [AugurProject/augur-core#126]
This creates many potential attack vectors.  One example is someone creating an event with 9 outcomes where the 9th outcome is by far the most likely outcome.  In this scenario, users won't be able to retrieve their rewards.

I believe that the only path to creating events _right now_ is via `createEvent`.  However, to avoid future risks (as things change over time) the validation should _also_ occur in the events contract.  This not only makes it clear to readers that events are constrained to 8 outcomes, it also makes it so someone can't introduce a subtle attack vector by allowing for an event to be created with an unbounded number of outcomes.

For a system as complex as Augur, I believe that this type of validation should occur at all levels.  There is simply too much code to maintain a high confidence that _all_ paths are validated the same way.  Validation _definitely_ should happen near the entrypoints.  However, the entrypoints are currently quite disconnected in some cases from the final location and sometimes there are multiple entry points that result internal initialization.  Ensuring these things are all in sync is a task that is very prone to failure.

Basically, with the exception of some simple helper methods, I would argue that all methods should fully validate their input, even if they are only being called internally.

If this were a language with `public`, `internal` and `private` then I would argue that only `public` and `internal` methods need full validation.  One option would be to add `private` member functions to Augur contracts by having an `if (self.sender !== self) throw()` line at the top of them.  `private` methods have a small enough scope of entry point that a human can reasonably ensure that all code paths are correctly validated, unlike `internal` methods that simply have too large of a potential call surface area.

**Comment by AlexeyAkhunov** [comment]
Interestingly, if there are maximum 8 outcomes, how would we handle French elections 2017, having 11 candidates? :)

**Comment by RonB7139** [comment]
generally, the market creator would set the top 7 most likely outcomes as
individual outcomes, and everyone else is in a "field" outcome -- meaning
anyone not mentioned in the specific individual outcomes of (the multiple
choice) market would be included in the field outcome...i.e. 1,2,3...7,
field = anyone else

**Comment by joeykrug** [comment]
Fixed
